### PR TITLE
[elasticsearch] add podAnnotations value

### PR DESCRIPTION
**Pull request by jslusher**
This creates a `podAnnotations` value for the client, master and data
templates. This will enable downstream charts to add annotations to Pods
created by this chart.

### CODE REVIEW

**Comment by jslusher** [comment]
I signed it!

File: incubator/elasticsearch/templates/client-deployment.yaml [hunk 1]
```diff
@@ -16,6 +16,10 @@ spec:
         app: {{ template "elasticsearch.name" . }}
         component: "{{ .Values.client.name }}"
         release: {{ .Release.Name }}
+      annotations:
```
Comment (mattfarina) [inline incubator/elasticsearch/templates/client-deployment.yaml]: Can you move the `annotation` inside the if statement. That way it's not in the generated yaml if it has no content. There are a few places to make this change.

**Review by mattfarina — CHANGES_REQUESTED** [review]

**Review by mattfarina — APPROVED** [review]

**Comment by mikn** [comment]
This looks good to me. :)

**Pull request merged**